### PR TITLE
rm adapter objects dummies

### LINKED ISSUE #770: Deprecation: remove adapter.objects and adater.states dummy objects

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #770

### CODE REVIEW

**Comment by foxriver76** [comment]
@Apollon77 please grep first if still used anywhere

File: packages/adapter/lib/adapter/adapter.js [hunk 1]
```diff
@@ -1617,7 +1613,7 @@ function Adapter(options) {
                     // if this.aliases is empty, or no target found its a new alias
                     let isNewAlias = true;
 
-                    Object.keys(this.aliases).forEach(sourceId => {
+                    for (const sourceId of Object.keys(this.aliases)) {
                         const alias = this.aliases[sourceId];
```
Comment (AlCalzone) [inline packages/adapter/lib/adapter/adapter.js:1617]:
```suggestion
                    for (const [sourceId, alias] of Object.entries(this.aliases)) {
```

File: packages/adapter/lib/adapter/adapter.js [hunk 2]
```diff
@@ -7772,23 +7756,22 @@ function Adapter(options) {
             this.aliasPatterns = this.aliasPatterns.filter(pattern => pattern !== aliasPattern);
 
             if (aliasPattern) {
-                Object.keys(this.aliases).forEach(sourceId => {
+                for (const sourceId of Object.keys(this.aliases)) {
```
Comment (AlCalzone) [inline packages/adapter/lib/adapter/adapter.js:7759]: Same here, this also avoids repeating `this.aliases[sourceId]`.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
We should think about making `this.aliases` a `Map` which allows for more performant accesses/length checks

**Comment by AlCalzone** [comment]
For some benchmarks:

even for small objects (100 entries), map.size is ~4x faster than Object.keys(...).length:
https://jsben.ch/CLIVp

bump the size up to 100000 and map.size is ~12000x faster
https://jsben.ch/6WZsu

Iterating looks like this map > Object.keys >> Object.entries
(actually the last bit is suprising)
https://jsben.ch/XG63T

Element access is approx. equal

**Comment by foxriver76** [comment]
MAkes sense to switch for me. But lets not do it in this PR, there are already some unrelated code changes 😄

**Comment by AlCalzone** [comment]
There are probably a **BUNCH** of locations where a map would be beneficial.

**Comment by Apollon77** [comment]
LGTM

**Pull request merged**